### PR TITLE
Expand failure tracking

### PR DESCRIPTION
**Pull request by nickbrowne**
Unfortunately I didn't realise that a bunch of tests weren't doing anything when I started writing this so I  don't actually know if it's going to work as intended.

Can't seem to find a way to emit an event to lita via the shell adapter, so I'll do it live and report back :fire:

### CODE REVIEW

File: spec/days_since_master_failure_spec.rb [hunk 1]
```diff
@@ -16,53 +16,79 @@
...
+            expect(robot).to have_received(:send_message).with(
+              an_instance_of(Lita::Source),
+              "tc is 2 days without a failure. The current record is 3. Aim for the top!"
+            )
+          end
         end
 
-        it "should send a message" do
-          handler.timestamp_failure(payload)
-          expect(robot).to_not have_received(:send_message).with(
```
Comment (nickbrowne) [inline spec/days_since_master_failure_spec.rb:62]: These are the tests that weren't doing anything, the negative assertions were always passing, even though the code they appeared to be exercising wasn't even being invoked.

Comment (tiagoamaro) [reply spec/days_since_master_failure_spec.rb:62]: > ...the negative assertions were always passing...

Negative assertions `¯\_(ツ)_/¯`

**Comment by tiagoamaro** [comment]
> Can't seem to find a way to emit an event to lita via the shell adapter, so I'll do it live and report back 🔥

Does this mean this PR is still a WIP? Checked out locally (specs) and it seems OK for me.

File: spec/days_since_master_failure_spec.rb (hunk 1 above)
Comment (yob) [reply spec/days_since_master_failure_spec.rb:62]: sheeet. Sorry!

**Comment by nickbrowne** [comment]
> Does this mean this PR is still a WIP? Checked out locally (specs) and it seems OK for me.

Yep, though I've actually shipped it already to test if it actually works the way I expect. Just observing what tcbot does until Monday :)

File: spec/days_since_master_failure_spec.rb (hunk 1 above)
Comment (nickbrowne) [reply spec/days_since_master_failure_spec.rb:62]: You tricked me good!

**Comment by nickbrowne** [comment]
Going to start this again because it seems to have broken things, this time by starting with backporting fixes for the broken tests to current master, closing this for now.

**Pull request closed without merging**